samples/bluetooth/bap_broadcast_sink/src/lc3.c

```diff
@@ -189,33 +229,11 @@ static size_t decode_frame_block(struct lc3_data *data, size_t frame_cnt)
 		 */
 		if (decode_frame(data, frame_cnt + decoded_frames)) {
 			decoded_frames++;
-
+			if (IS_ENABLED(CONFIG_USE_CODEC_AUDIO_OUTPUT)) {
+				codec_add_frame(stream, i, data->ts);
+			}
 			if (IS_ENABLED(CONFIG_USE_USB_AUDIO_OUTPUT)) {
-				enum bt_audio_location chan_alloc;
-				int err;
-
-				err = get_lc3_chan_alloc_from_index(stream, i, &chan_alloc);
-				if (err != 0) {
-					/* Not suitable for USB */
-					continue;
-				}
-
-				/* We only want to left or right from one stream to USB */
-				if ((chan_alloc == BT_AUDIO_LOCATION_FRONT_LEFT &&
-				     stream != usb_left_stream) ||
-				    (chan_alloc == BT_AUDIO_LOCATION_FRONT_RIGHT &&
-				     stream != usb_right_stream)) {
-					continue;
-				}
-
-				/* TODO: Add support for properly support the presentation delay.
-				 * For now we just send audio to USB as soon as we get it
-				 */
-				err = usb_add_frame_to_usb(chan_alloc, lc3_rx_buf,
-							   sizeof(lc3_rx_buf), data->ts);
-				if (err == -EINVAL) {
-					continue;
-				}
+				usb_add_frame(stream, i, data->ts);
```

Comment:
Both `usb_add_frame` and `codec_add_frame` may return errors that aren't handled here